app/javascript/src/useHiddenFields.js

```diff
@@ -0,0 +1,37 @@
+import { useCallback, useEffect, useState } from "react";
+import * as storage from "./localStorage";
+
+export const useHiddenFields = (storageKey, defaultHiddenFields) => {
+  const [hiddenFields, setHiddenFields] = useState(defaultHiddenFields || []);
```

Comment:
Here I started with setting the default value on the function parameter (`(storageKey, defaultHiddenFields = [])`), but that caused a render loop. Likely the array is new each time, so it breaks `===`.